services/storage/src/simcore_service_storage/s3_client.py

```diff
@@ -303,16 +275,22 @@ async def delete_files_in_path(self, bucket: S3BucketName, *, prefix: str) -> No
 
         # NOTE: deletion of objects is done in batches of max 1000 elements,
         # the maximum accepted by the S3 API
-        async for s3_objects in self.list_all_objects_gen(
-            bucket,
-            prefix=prefix,
-            max_yield_result_size=_DELETE_OBJECTS_MAX_ACCEPTED_ELEMENTS,
-        ):
-            if objects_to_delete := [f["Key"] for f in s3_objects if "Key" in f]:
-                await self.client.delete_objects(
-                    Bucket=bucket,
-                    Delete={"Objects": [{"Key": key} for key in objects_to_delete]},
-                )
+        with log_context(_logger, logging.INFO, "deleting objects", log_duration=True):
+            object_groups_to_delete: list[list[str]] = []
+            async for s3_objects in self.list_all_objects_gen(bucket, prefix=prefix):
```

Comment:
If i understand this right:
-  `_list_objects_v2_paginated_gen` return a list of object that fit in one page and you forward this result here i.e. `list_all_objects_gen` is also iterating pages.
- Then here, you basically accumulate all pages  to delete into `object_groups_to_delete` 
- Finally you batch-delete every page in parallel with `gather`

Why you are not taking advantages of the iterator and calling delete as they page comes?
Why you are accumulating instead all the pages produced by the iterator?
In principle, the previous solution sounds more logic to me. What am I missing here?
thx in advance for your answer.

Reply:
I am also interested in understanding the issue here. 
I agree with @pcrespov that accumulating all of them here might be massive. You could actually go page by page and delete. Or even start multiple deletion tasks if that is really an issue. I think it is ok if it takes some time, as long as it is not taking minutes.

Reply:
let me batch them page by page and see the difference, if it's minor, I will let it sequential, less load on overall resources.